tests/integration/test_docker_integration.py

```diff
@@ -2,18 +2,26 @@
 import logging
 import pytest
 import time
+import shlex
 
 
 log = logging.getLogger(__name__)
 
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test):
-    bundle = ops_test.render_bundle(
-        "tests/data/bundle.yaml", docker_charm=await ops_test.build_charm(".")
-    )
+    log.info("Build Charm...")
+    charm = await ops_test.build_charm(".")
+
+    log.info("Build Bundle...")
+    bundle = ops_test.render_bundle("tests/data/bundle.yaml", docker_charm=charm)
+
+    log.info("Deploy Bundle...")
+    model = ops_test.model_full_name
+    cmd = f"juju deploy -m {model} {bundle}"
+    rc, stdout, stderr = await ops_test.run(*shlex.split(cmd))
+    assert rc == 0, f"Bundle deploy failed: {(stderr or stdout).strip()}"
 
```

Comment:
what's the reasoning for not deploying the bundle with pytest operator

Reply:
it seems `ops_test.deploy_bundle` requires another snap package in the environment called `juju-bundle` which relies on a library `rust-libjuju` which is out of date in supporting any ubuntu series past [eoan](https://github.com/knkski/rust-libjuju/issues/3).

this is the best i've got access to at the moment.